### PR TITLE
Adap 1126/add test for pr 9682 in core

### LINKED ISSUE #1126: [Bug]  Add test for 9682 in core against 1.7
### Is this a new bug in dbt-bigquery?

- [X] I believe this is a new bug in dbt-bigquery
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

No test. Not instantiating frozensets right.

### Expected Behavior

Test the working changes.

### Steps To Reproduce

See [this core PR](https://github.com/dbt-labs/dbt-core/pull/9682)

### Relevant log output

_No response_

### Environment

```markdown
- OS:
- Python:
- dbt-core:
- dbt-bigquery:
```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by VersusFacit**
resolves #1126 

### Checklist

- [X] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [X] I have run this code in development and it appears to resolve the stated issue
- [X] This PR includes tests, or tests are not required/relevant for this PR
- [X] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

File: dbt/adapters/bigquery/relation.py
Comment (mikealfare) [inline dbt/adapters/bigquery/relation.py:1]: I would leave this as-is, unless it's failing.

File: dev-requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # install latest changes in dbt-core
 # TODO: how to automate switching from develop to version branches?
-git+https://github.com/dbt-labs/dbt-core.git@1.7.latest#egg=dbt-core&subdirectory=core
+git+https://github.com/dbt-labs/dbt-core.git@ADAP-9681/fix_renamed_relations_override_pre_1.8#egg=dbt-core&subdirectory=core
```
Comment (mikealfare) [inline dev-requirements.txt:3]: Leaving a note to point this back to `1.7.latest` before merging.

File: dbt/adapters/bigquery/relation.py
Comment (VersusFacit) [reply dbt/adapters/bigquery/relation.py:1]: It's failing otherwise.

**Comment by VersusFacit** [comment]
Not backporting this

**Pull request closed without merging**